classes/Hook/HookActionCarrierProcess.php

```diff
@@ -78,10 +78,12 @@ protected function getUniversalAnalytics(string $carrierName)
 
     protected function getGoogleAnalytics4(string $carrierName)
     {
-        return 'gtag("event", "add_shipping_info", {
-            currency: "' . $this->context->currency->iso_code . '",
-            value: ' . $this->context->cart->getCartTotalPrice() . ',
-            shipping_tier: "' . $carrierName . '"
-          });';
+        $eventData = [
+            'currency' => $this->context->currency->iso_code,
+            'value' => (float) $this->context->cart->getSummaryDetails()['total_price'],
+            'shipping_tier' => $carrierName,
+        ];
+
+        return 'gtag("event", "add_shipping_info", ' . json_encode($eventData, JSON_UNESCAPED_UNICODE) . ');';
```

Comment:
It's a nitpick but I'd use sprintf for readability:

```suggestion
        return sprintf(
            'gtag("event", "add_shipping_info", %s);',
            json_encode($eventData, JSON_UNESCAPED_UNICODE)
        );
```